ledger/src/blockstore.rs

```diff
@@ -3501,6 +3501,66 @@ impl Blockstore {
         Ok((entries, num_shreds, slot_meta.is_full()))
     }
 
+    pub fn get_chunked_slot_entries_in_block(
+        &self,
+        slot: Slot,
+        start_index: u64,
+        allow_dead_slots: bool,
+        mut callback: impl FnMut(
+            (Vec<Entry>, u64, bool),
+        ) -> std::result::Result<(), BlockstoreProcessorError>,
+    ) -> std::result::Result<(), BlockstoreProcessorError> {
+        if self.is_dead(slot) && !allow_dead_slots {
+            Err(BlockstoreError::DeadSlot)?;
+        }
+        let slot_meta = self.meta_cf.get(slot)?.unwrap();
+        assert!(!slot_meta
+            .completed_data_indexes
+            .contains(&(slot_meta.consumed as u32)));
+
+        let start_index = start_index as u32;
+        for (start, end) in slot_meta
+            .completed_data_indexes
+            .range(start_index..slot_meta.consumed as u32)
+            .scan(start_index, |begin, index| {
+                let out = (*begin, *index);
+                *begin = index + 1;
+                Some(out)
+            })
+        {
+            let keys = (start..=end).map(|index| (slot, u64::from(index)));
+            let range_shreds = self
+                .data_shred_cf
+                .multi_get_bytes(keys)
```

Comment:
i wonder how much overhead it incurs here instead of one-giant invocation of `.multi_get_bytes()`.